core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -786,6 +746,68 @@ object Pull extends PullLowPriority {
         }
       }
 
+      def goFlatMapOut[Y](fmout: FlatMapOutput[G, Y, X], view: View[G, X, Unit]): F[End] = {
+        val runr = new BuildR[G, Y, End]
+        val stepRunR = new FlatMapR(view, fmout.fun)
+        F.unit >> // The F.unit is needed because otherwise an stack overflow occurs.
```

Comment:
Hm, is this `F.unit >> go` safe?

Reply:
What part of `safe` are you worried about?  The reason for adding this `F.unit >> go` is that, without it, there is a stack overflow in the evaluation of some tests.